ee/control/consumers/tokenconsumer/ingest_token_consumer.go

```diff
@@ -0,0 +1,41 @@
+package tokenconsumer
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/kolide/launcher/pkg/agent/types"
+)
+
+const IngestSubsystem = "ingest"
+
+var ObservabilityIngestTokenKey = []byte("observability_ingest_server_token")
+
+type IngestTokenConsumer struct {
+	store types.KVStore
+}
+
+type traceSubsystemConfig struct {
+	IngestToken string `json:"ingest_token"`
+}
+
+func NewIngestTokenConsumer(store types.KVStore) *IngestTokenConsumer {
```

Comment:
if the idea of this consumer is that it is just for bearer auth tokens, I think it should be a little more generally names. Like `tokenconsumer` or `authtokenconsumer`

```suggestion
func NewTokenConsumer(store types.KVStore) *tokenConsumer {
```

Reply:
Also covering https://github.com/kolide/launcher/pull/1226#discussion_r1230159791 and https://github.com/kolide/launcher/pull/1226#discussion_r1230153466 --

I was waffling about naming for this whole package mostly because I'm unsure about how it would expand. Will we want it to handle other tokens used for other things (so, subsystem `auth_tokens`, consumer `TokenConsumer`), or will we want it to handle other ingest-related items (so, subsystem `ingest`, consumer `IngestConfig(?)Consumer`)?

All the other ingest-related control server items ended fitting up better into the agent flags subsystem, so I don't really have any plans to put other ingest-related items in this subsystem's update data. But I also don't know if we plan to ship other tokens, either. Do you have thoughts/plans one way or the other? If not, maybe the clearest option would be to go super-specific (subsystem `ingest_auth_tokens`, consumer `IngestAuthTokensConsumer`)?